<launch>
    <param name="/use_sim_time" value="false"/>
    <arg name="USE_IMU"  default="0"/>
    <arg name="URL_CONFIG_FOLDER"  default="/home/sivvon/Desktop/Junseok_backup_0410/knu/scanning_deep/catkin_ws_scanning_ekf_ver2/config"/>
    <arg name="config_file"  default="$(arg URL_CONFIG_FOLDER)/kitti_config_03/parking_crvlResize.yaml"/>
    <rosparam command="load" file="$(arg URL_CONFIG_FOLDER)/kitti_config_03/imu_vo_cfg.yaml"/>
        
    <node name="rvizvisualisation" pkg="rviz" type="rviz" output="screen" args="-d $(arg URL_CONFIG_FOLDER)/knu_rviz_config.rviz" />

    <node name="vins_estimator" pkg="vins" type="vins_node" output="screen">
        <param name="config_file"   type="string" value="$(arg config_file)"/>
    </node>

    <node name="rosLidarNode" pkg="vins" type="rosLidarNode" output="screen">        
        <param name="USE_IMU"      type="int"    value="$(arg USE_IMU)"/>
        <param name="config_file"   type="string" value="$(arg config_file)"/>d
    </node>

    <!--node pkg="imu_complementary_filter" type="complementary_filter_node" name="complementary_filter_node">
    	<remap from="/imu/data_raw" to="/imu/data"/>
	    <remap from="/imu/data" to="/imu_data"/>    	
    </node>
    <node pkg="vins" type="sync_node" name="sync_node"/>
    <node pkg="vins" type="cvt_odometry" name="cvt_odometry"/>    
    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
      <remap from="odom"     		to="/gpo"/>
      <remap from="vo"       		to="/sync_odom"/>
      <remap from="imu_data" 		to="/sync_imu"/>
      <param name="output_frame" 	value="world"/>
      <param name="base_footprint_frame" value="body"/>
      <param name="freq" 		    value="50.0"/>
      <param name="sensor_timeout" 	value="1.0"/>
      <param name="odom_used" 	    value="false"/>
      <param name="imu_used" 		value="true"/>
      <param name="vo_used" 		value="true"/>
      <param name="gps_used" 		value="false"/>
      <param name="debug" 	 	    value="false"/>
      <param name="self_diagnose" 	value="false"/>
    </node-->
    
</launch>
